examples/cord_lc/main.c

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2018 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     examples
+ * @{
+ *
+ * @file
+ * @brief       CoRE Resource Directory lookup (cord_lc) example
+ *
+ * @author      Aiman Ismail <muhammadaimanbin.ismail@haw-hamburg.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+
+#include "shell.h"
+#include "msg.h"
+
+#define MAIN_QUEUE_SIZE     (8)
+static msg_t _main_msg_queue[MAIN_QUEUE_SIZE];
+
+int main(void)
+{
+    /* we need a message queue for the thread running the shell in order to
+     * receive potentially fast incoming networking packets */
+    msg_init_queue(_main_msg_queue, MAIN_QUEUE_SIZE);
```

Comment:
I removed that section and tested it by pinging to itself and got a FAILED_ASSERTION. So maybe it is still needed. Note: I'm using master branch from Oct 21.

Reply:
> If I remember correctly, @miri64 told me that this is not needed anymore, right @miri64?!

If you are not using the message queue of the thread for netapi, that is correct, yes. Since `sock`s use their own message queues (mbox) they don't need it.

Reply:
(`sock` uses `mbox` since its inception, so your version *should* work without if *if* you are not using vanilla `gnrc_netapi`).

Reply:
(it is however still needed for the `ping6` command, if you include that)